app/lib/pds/helpers/data_helpers.rb

```diff
@@ -37,6 +37,17 @@ def update_or_set_new_timestamps!(entity_type, resources)
           end
         end
       end
+
+      def set_user_tokens!(user_candidates)
+        user_candidates.each do |candidate|
+          uuid = SecureRandom.uuid.upcase
+          candidate_initial = candidate['username'].chars.first(2).join.upcase
+          seconds_until_eod = Time.now.seconds_until_end_of_day
+          new_token = "#{seconds_until_eod}-#{uuid}-#{candidate_initial}"
```

Comment:
**Note for the reviewer:**

`SecureRandom` is supposed to be pretty solid in terms of generating random digits (UUID in this case)

But I wanted to add two more variables to the token in case the _random_ doesn't behave well in bulk operations

Reply:
I think we should be fine with just `SecureRandom.uuid.upcase`.

Out of principle (information leakage, need-to-know) we probably shouldn't include any potentially user-identifying information (like initial username characters) in the string. `Time.now.seconds_until_end_of_day` will probably all be the same for a batch of users ('cause computers are fast), and even if it's not the same it's still predictable.

SecureRandom is almost certainly already doing way more to make this unpredictable than anything we can add to it postfacto.